### PR TITLE
restore_wallet rpc

### PR DESCRIPTION
**Pull request by cryptoseyed**
Allow to restore a wallet via RPC with the seed word list. 

- supported word lists of length: 12, 14, 24, 25 or 26

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -110,3 +110,6 @@ local.properties
 .idea/
 
 /testnet
+
+# VS Code
+.vscode
```
Comment (psychocrypt) [inline .gitignore]: please add a empty line before EOF

**Comment by psychocrypt** [comment]
I added a small description to you pull request.

File: src/wallet/wallet_rpc_server.cpp [hunk 2]
```diff
@@ -2431,6 +2431,97 @@ bool wallet_rpc_server::on_create_wallet(const wallet_rpc::COMMAND_RPC_CREATE_WA
...
+	po::variables_map vm2;
+
+	{
+		po::options_description desc("dummy");
+		const command_line::arg_descriptor<std::string, true> arg_password = {"password", "password"};
+		const char *argv[4];
+		int argc = 3;
+		argv[0] = "wallet-rpc";
+		argv[1] = "--password";
+		argv[2] = req.password.c_str();
```
Comment (psychocrypt) [inline src/wallet/wallet_rpc_server.cpp]: Only a note: This construct is risky if `req` is destroyed and `argv` is used after that point the pointer in `argv[2]` would be invalid. This is not the case in this code snipped but I would point to this risky snipped.

File: src/wallet/wallet_rpc_server.cpp [hunk 3]
```diff
@@ -2431,6 +2431,97 @@ bool wallet_rpc_server::on_create_wallet(const wallet_rpc::COMMAND_RPC_CREATE_WA
...
+	wallet->set_refresh_from_block_height(req.refresh_start_height);
+
+	try
+	{
+		if(decode_14)
+			wallet->generate_new(wallet_file, req.password, &seed_14, seed_extra, false);
+		else
+			wallet->generate_legacy(wallet_file, req.password, seed_25, false);
+
+		if(!wallet)
```
Comment (psychocrypt) [inline src/wallet/wallet_rpc_server.cpp]: I do not understand this check: wallet is a unique pointer and use many time with `->` before. If wallet is a `nullptr` tha nwe did many invalid memory accesses before. Could it be that this  check must be in line `2496`

File: src/wallet/wallet_rpc_server_commands_defs.h [hunk 4]
```diff
@@ -1624,6 +1624,29 @@ struct COMMAND_RPC_CREATE_WALLET
 	};
 };
 
+struct COMMAND_RPC_RESTORE_WALLET
+{
+	struct request
+	{
+		std::string filename;
+		std::string password;
```
Comment (psychocrypt) [inline src/wallet/wallet_rpc_server_commands_defs.h]: @fireice-uk A note: we transfer here a plain test password, thats not good.

Comment (fireice-uk) [reply src/wallet/wallet_rpc_server_commands_defs.h]: We give the user a choice to supply it from command line - this is similar to how wallet creation works

File: src/wallet/wallet_rpc_server.cpp (hunk 3 above)
Comment (cryptoseyed) [reply src/wallet/wallet_rpc_server.cpp]: I think you are correct and I should move it to `2496`

File: .gitignore (hunk 1 above)
Comment (cryptoseyed) [reply .gitignore]: Done.

**Pull request merged**